File: ruby/test/get_dressed_test.rb
#!/usr/bin/ruby
Dir[File.dirname(__FILE__) + './lib/*.rb'].each {|file| require file }

puts "Test: ", "  input = '5 1', output = 'socks, fail'"
num = '5 1'
puts '--------------------RESULT---------------------------'
if processTheDress(num) == 'socks, fail'
  puts "  output = " + processTheDress(num)
  puts "  \033[32mTest Passed √\033[0m"
else
  puts "  \033[31m! Test Fail ×\033[0m"
  puts "  The output is: " + processTheDress(num)
end
puts '-----------------------------------------------------'

puts ''

puts "Test: ", "  input = '4 1', output = 'fail'"
num = '4 1'
puts '--------------------RESULT---------------------------'
if processTheDress(num) == 'fail'
  puts "  output = " + processTheDress(num)
  puts "  \033[32mTest Passed √\033[0m"
else
  puts "  \033[31m! Test Fail ×\033[0m"
  puts "  The output is: " + processTheDress(num)
end
puts '-----------------------------------------------------'

puts ''

puts "Test: ", "  input = '3 4 2 5 6', output = 'shirt, fail'"
num = '3 4 2 5 6'
puts '--------------------RESULT---------------------------'
if processTheDress(num) == 'shirt, fail'
  puts "  output = " + processTheDress(num)
  puts "  \033[32mTest Passed √\033[0m"
else
  puts "  \033[31m! Test Fail ×\033[0m"
  puts "  The output is: " + processTheDress(num)
end
puts '-----------------------------------------------------'

puts ''

puts "Test: ", "  input = '5 3 4 2 6', output = 'socks, shirt, fail'"
num = '5 3 4 2 6'
puts '--------------------RESULT---------------------------'
if processTheDress(num) == 'socks, shirt, fail'
  puts "  output = " + processTheDress(num)
  puts "  \033[32mTest Passed √\033[0m"
else
  puts "  \033[31m! Test Fail ×\033[0m"
  puts "  The output is: " + processTheDress(num)
end
puts '-----------------------------------------------------'

puts ''

puts "Test: ", "  input = '5 2 1 4 3 6', output = 'socks, pants, fail'"
num = '5 2 1 4 3 6'
puts '--------------------RESULT---------------------------'
if processTheDress(num) == 'socks, pants, fail'
  puts "  output = " + processTheDress(num)
  puts "  \033[32mTest Passed √\033[0m"
else
  puts "  \033[31m! Test Fail ×\033[0m"
  puts "  The output is: " + processTheDress(num)
end
puts '-----------------------------------------------------'

puts ''

puts "Test: ", "  input = '3 2 5 1', output = 'shirt, pants, socks, hat, fail'"
num = '3 2 5 1'
puts '--------------------RESULT---------------------------'
if processTheDress(num) == 'shirt, pants, socks, hat, fail'
  puts "  output = " + processTheDress(num)
  puts "  \033[32mTest Passed √\033[0m"
else
  puts "  \033[31m! Test Fail ×\033[0m"
  puts "  The output is: " + processTheDress(num)
end
puts '-----------------------------------------------------'

puts ''

puts "Test: ", "  input = '5 3 2 6 4', output = 'socks, shirt, pants, fail'"
num = '5 3 2 6 4'
puts '--------------------RESULT---------------------------'
if processTheDress(num) == 'socks, shirt, pants, fail'
  puts "  output = " + processTheDress(num)
  puts "  \033[32mTest Passed √\033[0m"
else
  puts "  \033[31m! Test Fail ×\033[0m"
  puts "  The output is: " + processTheDress(num)
end
puts '-----------------------------------------------------'

puts ''

puts "Test: ", "  input = '5 3 2 4 6', output = 'socks, shirt, pants, shoes, leave'"
num = '5 3 2 4 6'
puts '--------------------RESULT---------------------------'
if processTheDress(num) == 'socks, shirt, pants, shoes, leave'
  puts "  output = " + processTheDress(num)
  puts "  \033[32mTest Passed √\033[0m"
else
  puts "  \033[31m! Test Fail ×\033[0m"
  puts "  The output is: " + processTheDress(num)
end
puts '-----------------------------------------------------'

puts ''

puts "Test: ", "  input = '3 5 1 2 4 6', output = 'shirt, socks, hat, pants, shoes, leave'"
num = '3 5 1 2 4 6'
puts '--------------------RESULT---------------------------'
if processTheDress(num) == 'shirt, socks, hat, pants, shoes, leave'
  puts "  output = " + processTheDress(num)
  puts "  \033[32mTest Passed √\033[0m"
else
  puts "  \033[31m! Test Fail ×\033[0m"
  puts "  The output is: " + processTheDress(num)
end
puts '-----------------------------------------------------'

puts ''

puts "Test: ", "  input = '3 2 5 4 1 6', output = 'shirt, pants, socks, shoes, hat, leave'"
num = '3 2 5 4 1 6'
puts '--------------------RESULT---------------------------'
if processTheDress(num) == 'shirt, pants, socks, shoes, hat, leave'
  puts "  output = " + processTheDress(num)
  puts "  \033[32mTest Passed √\033[0m"
else
  puts "  \033[31m! Test Fail ×\033[0m"
  puts "  The output is: " + processTheDress(num)
end
puts '-----------------------------------------------------'

puts ''

puts "Test: ", "  input = '3 5 2 4 1 6', output = 'shirt, socks, pants, shoes, hat, leave'"
num = '3 5 2 4 1 6'
puts '--------------------RESULT---------------------------'
if processTheDress(num) == 'shirt, socks, pants, shoes, hat, leave'
  puts "  output = " + processTheDress(num)
  puts "  \033[32mTest Passed √\033[0m"
else
  puts "  \033[31m! Test Fail ×\033[0m"
  puts "  The output is: " + processTheDress(num)
end
puts '-----------------------------------------------------'

puts ''

puts "Test: ", "  input = '5 2 3 4 1 6', output = 'socks, pants, shirt, shoes, hat, leave'"
num = '5 2 3 4 1 6'
puts '--------------------RESULT---------------------------'
if processTheDress(num) == 'socks, pants, shirt, shoes, hat, leave'
  puts "  output = " + processTheDress(num)
  puts "  \033[32mTest Passed √\033[0m"
else
  puts "  \033[31m! Test Fail ×\033[0m"
  puts "  The output is: " + processTheDress(num)
end
puts '-----------------------------------------------------'

File: python/test/get_dressed_test.py
# -*- coding: utf-8 -*-
import sys
sys.path.append('../')
from lib.get_dressed import *

num = '5 1'
print ('')
print ("Test: ")
print ("  input = '5 1', output = 'socks, fail'")
assert processTheDress(num) == 'socks, fail'
print ("  \033[32mTest Passed √\033[0m")
print ('-----------------------------------------------------')

num = '4 1'
print ('')
print ("Test: ")
print ("  input = '4 1', output = 'fail'")
assert processTheDress(num) == 'fail'
print ("  \033[32mTest Passed √\033[0m")
print ('-----------------------------------------------------')

num = '3 4 2 5 6'
print ('')
print ("Test: ")
print ("  input = '3 4 2 5 6', output = 'shirt, fail'")
assert processTheDress(num) == 'shirt, fail'
print ("  \033[32mTest Passed √\033[0m")
print ('-----------------------------------------------------')

num = '5 3 4 2 6'
print ('')
print ("Test: ")
print ("  input = '5 3 4 2 6', output = 'socks, shirt, fail'")
assert processTheDress(num) == 'socks, shirt, fail'
print ("  \033[32mTest Passed √\033[0m")
print ('-----------------------------------------------------')

num = '5 2 1 4 3 6'
print ('')
print ("Test: ")
print ("  input = '5 2 1 4 3 6', output = 'socks, pants, fail'")
assert processTheDress(num) == 'socks, pants, fail'
print ("  \033[32mTest Passed √\033[0m")
print ('-----------------------------------------------------')

num = '3 2 5 1'
print ('')
print ("Test: ")
print ("  input = '3 2 5 1', output = 'shirt, pants, socks, hat, fail'")
assert processTheDress(num) == 'shirt, pants, socks, hat, fail'
print ("  \033[32mTest Passed √\033[0m")
print ('-----------------------------------------------------')

num = '5 3 2 6 4'
print ('')
print ("Test: ")
print ("  input = '5 3 2 6 4', output = 'socks, shirt, pants, fail'")
assert processTheDress(num) == 'socks, shirt, pants, fail'
print ("  \033[32mTest Passed √\033[0m")
print ('-----------------------------------------------------')

num = '5 3 2 4 6'
print ('')
print ("Test: ")
print ("  input = '5 3 2 4 6', output = 'socks, shirt, pants, shoes, leave'")
assert processTheDress(num) == 'socks, shirt, pants, shoes, leave'
print ("  \033[32mTest Passed √\033[0m")
print ('-----------------------------------------------------')

num = '3 5 1 2 4 6'
print ('')
print ("Test: ")
print ("  input = '3 5 1 2 4 6', output = 'shirt, socks, hat, pants, shoes, leave'")
assert processTheDress(num) == 'shirt, socks, hat, pants, shoes, leave'
print ("  \033[32mTest Passed √\033[0m")
print ('-----------------------------------------------------')

num = '3 2 5 4 1 6'
print ('')
print ("Test: ")
print ("  input = '3 2 5 4 1 6', output = 'shirt, pants, socks, shoes, hat, leave'")
assert processTheDress(num) == 'shirt, pants, socks, shoes, hat, leave'
print ("  \033[32mTest Passed √\033[0m")
print ('-----------------------------------------------------')

num = '3 5 2 4 1 6'
print ('')
print ("Test: ")
print ("  input = '3 5 2 4 1 6', output = 'shirt, socks, pants, shoes, hat, leave'")
assert processTheDress(num) == 'shirt, socks, pants, shoes, hat, leave'
print ("  \033[32mTest Passed √\033[0m")
print ('-----------------------------------------------------')

num = '5 2 3 4 1 6'
print ('')
print ("Test: ")
print ("  input = '5 2 3 4 1 6', output = 'socks, pants, shirt, shoes, hat, leave' ")
assert processTheDress(num) == 'socks, pants, shirt, shoes, hat, leave'
print ("  \033[32mTest Passed √\033[0m")
print ('-----------------------------------------------------')